appinventor/appengine/src/com/google/appinventor/client/editor/simple/SimpleVisibleComponentsPanel.java

```diff
@@ -83,56 +88,83 @@ public void onValueChange(ValueChangeEvent<Boolean> event) {
-                SettingsConstants.YOUNG_ANDROID_SETTINGS_PHONE_TABLET));
-        checkboxPhoneTablet.setValue(showPhoneTablet);
-        changeFormPreviewSize(showPhoneTablet);
+        String val = projectEditor.getProjectSettingsProperty(SettingsConstants.PROJECT_YOUNG_ANDROID_SETTINGS,
+            SettingsConstants.YOUNG_ANDROID_SETTINGS_PHONE_TABLET);
+
+        int idx = 0;
+        int width = 320;
+        int height = 505;
+
+        if (val.equals("True")) {
+          idx = 1;
+          width = drop_lst[idx][0];
+          height = drop_lst[idx][1];
+        }
+
+        String[] parts = val.split(",");
+        if (parts.length == 3) {
+          idx = Integer.parseInt(parts[0]);
+          width = Integer.parseInt(parts[1]);
+          height = Integer.parseInt(parts[2]);
+        }
+        listboxPhoneTablet.setItemSelected(idx, true);
+        changeFormPreviewSize(idx, width, height);
       }
     };
-    checkboxPhoneTablet.addValueChangeHandler(new ValueChangeHandler<Boolean>() {
+    listboxPhoneTablet.addItem("Phone size");
+    listboxPhoneTablet.addItem("Tablet size");
+    listboxPhoneTablet.addItem("Monitor size");
+    listboxPhoneTablet.addChangeHandler(new ChangeHandler() {
       @Override
-      public void onValueChange(ValueChangeEvent<Boolean> event) {
-          boolean isChecked = event.getValue(); // auto-unbox from Boolean to boolean
-          projectEditor.changeProjectSettingsProperty(
-              SettingsConstants.PROJECT_YOUNG_ANDROID_SETTINGS,
-              SettingsConstants.YOUNG_ANDROID_SETTINGS_PHONE_TABLET,
-              isChecked ? "True" : "False");
-          changeFormPreviewSize(isChecked);
-        }
+      public void onChange(ChangeEvent event) {
+        int idx = listboxPhoneTablet.getSelectedIndex();
+        int width = drop_lst[idx][0];
+        int height = drop_lst[idx][1];
+        String val = Integer.toString(idx) + "," + Integer.toString(width) + "," + Integer.toString(height);
+        // here, we can change settings by putting val into it
+        projectEditor.changeProjectSettingsProperty(SettingsConstants.PROJECT_YOUNG_ANDROID_SETTINGS,
+            SettingsConstants.YOUNG_ANDROID_SETTINGS_PHONE_TABLET, val);
+        changeFormPreviewSize(idx, width, height);
+      }
     });
-    phoneScreen.add(checkboxPhoneTablet);
+    phoneScreen.add(listboxPhoneTablet);
 
     initWidget(phoneScreen);
   }
 
-  private void changeFormPreviewSize(boolean isChecked) {
-    if (form != null){
-      if (isChecked){
-        form.changePreviewSize(true);
-        checkboxPhoneTablet.setText(MESSAGES.previewPhoneSize());
-      }
-      else {
-        form.changePreviewSize(false);
-        checkboxPhoneTablet.setText(MESSAGES.previewTabletSize());
+  private void changeFormPreviewSize(int idx, int width, int height) {
+    if (form != null) {
+      form.changePreviewSize(width, height);
+      String info = "(" + height + "," + width + ")";
```

Comment:
Might want to have the first part of this be " (" to give some room between the message and the size.

Reply:
Done.